core/src/test/java/org/infinispan/expiration/impl/ClusterExpirationFunctionalTest.java

```diff
@@ -139,16 +164,40 @@ private void testLifespanExpiredEntryRetrieval(Cache<Object, String> primaryOwne
       }
    }
 
-   private MagicKey createKey(Cache<Object, String> primaryOwner, Cache<Object, String> backupOwner) {
-      if (cacheMode.isScattered()) {
-         return new MagicKey(primaryOwner);
+   private Object createKey(Cache<Object, String> primaryOwner, Cache<Object, String> backupOwner) {
+      if (storageType == StorageType.OBJECT) {
+         if (cacheMode.isScattered()) {
+            return new MagicKey(primaryOwner);
+         } else {
+            return new MagicKey(primaryOwner, backupOwner);
+         }
       } else {
-         return new MagicKey(primaryOwner, backupOwner);
+         // BINARY and OFF heap can't use MagicKey as they are serialized
```

Comment:
IMO `MagicKey` could do it, but instead of computing a hashCode and always returning it, it needs to generate a random value so that the hash of the serialized form has the required owners.

Reply:
Yes, I would have to create a MagicKey that when serialized mapped to the appropriate owners. It seemed better to not rely upon that and just use a simple number.

Reply:
Ok